custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/checkstyle/filters/MaxNumberOfViolationsInAFileFilter.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle.filters;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.Filter;
+
+/**
+ * @author Lyubomir Papazov - Initial contribution
+ *
+ *         This filter sets the maximum number of violations to be reported for
+ *         a single file for certain checks, specified in the setChecks method.
+ *
+ *         In order to use this module, the user must create a
```

Comment:
Is there a reason you did not included actual usage or a test?

Reply:
I just remember the other PR, but are there no other bindings who could profit? Checks for tabs maybe vs spaces maybe?

Reply:
This Filter was indeed created to bound the number of warnings for the WindowsLineEndings check. However, it turns out that the WindowsLineEnding check won't be useful.

Both the OnlyTabIndentationInXmlFilesCheck and FileTabCharacterCheck trigger errors and such a filter would make no difference for them.